module/Activity/src/Activity/Controller/AdminController.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: stefan
+ * Date: 19-7-15
+ * Time: 11:39
+ */
+
+namespace Activity\Controller;
+
+use Zend\Mvc\Controller\AbstractActionController;
+use Activity\Model\Activity;
+
+/**
+ * Controller for all administrative activity actions
+ */
+class AdminController extends AbstractActionController
+{
+    /**
+     * View the queue of not approved activities
+     */
+    public function queueAction()
+    {
+        $activityService = $this->getServiceLocator()->get('activity_service_activity');
+        $unapprovedActivities = $activityService->getUnapprovedActivities();
+        $approvedActivities = $activityService->getApprovedActivities();
+        $disapprovedActivities = $activityService->getDisapprovedActivities();
+
+        return [
+            'unapprovedActivities' => $unapprovedActivities,
+            'approvedActivities' => $approvedActivities,
+            'disapprovedActivities' => $disapprovedActivities
+        ];
+    }
+
+    /**
+     * View one activity.
+     */
+    public function viewAction()
+    {
+        $id = (int) $this->params('id');
+        $activityService = $this->getServiceLocator()->get('activity_service_activity');
+
+        /** @var $activity Activity*/
+        $activity = $activityService->getActivity($id);
+
+        if (is_null($activity)) {
+            return $this->notFoundAction();
+        }
+
+        return [
+            'activity' => $activity,
+        ];
+    }
+
+    /**
+     * Approve of an activity
+     */
+    public function approveAction()
+    {
+        return $this->setApprovalStatus('approve');
+    }
+
+    /**
+     * Disapprove an activity
+     */
+    public function disapproveAction()
+    {
+        return $this->setApprovalStatus('disapprove');
+    }
+
+    /**
+     * Reset the approval status of an activity
+     */
+    public function resetAction()
+    {
+        return $this->setApprovalStatus('reset');
+    }
+
+    /**
+     * Set the approval status of the activity requested
+     *
+     * @param $status
+     * @return array|\Zend\Http\Response
+     */
+    protected function setApprovalStatus($status)
+    {
+        $id = (int) $this->params('id');
+        $activityService = $this->getServiceLocator()->get('activity_service_activity');
+
+        /** @var $activity Activity*/
+        $activity = $activityService->getActivity($id);
+
+        if (is_null($activity)) {
+            return $this->notFoundAction();
+        }
+
+        switch ($status) {
+            case 'approve':
+                $activityService->approve($activity);
+                break;
+            case 'disapprove':
+                $activityService->disapprove($activity);
+                break;
+            case 'reset':
+                $activityService->reset($activity);
+                break;
+            default:
+                throw new \InvalidArgumentException('No sutch status ' . $status);
```

Comment:
this is not text that is returned to the user, so it does not have to be translated.

Also for the exception, I assume that will never reach the end user. and we are able to understand Dutch